msg.go

```diff
@@ -364,11 +367,21 @@ Loop:
 						for _, r := range bufs {
 							s = append(s, r)
 						}
+						maxLen += 3 // unprintable char is 3 extra bytes
 					} else {
 						s = append(s, b)
 					}
 				}
 			}
+			// never exceed the allowed label count lenght (63) 
+			if labels >= 63 { 
+				return "", lenmsg, &Error{err: "name exceeds 63 labels"}
+			}
+			labels += 1
+			// never exceed the allowed doman name length (255 octets) 
+			if len(s) >= maxLen {
+				return "", lenmsg, &Error{err: "name exceeded allowed 255 octets"}
```

Comment:
I agree with @andrewtj, this sounds a bit too corner case-y. Think the current check is fine.